packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -47,7 +47,7 @@ export interface TransactionsTrackerProps {
 }
 
 export interface TransactionsTrackerInternals {
-  transactionsSource$?: Observable<Cardano.TxAlonzo[]>;
+  transactionsSource$?: Observable<{ transactions: Cardano.TxAlonzo[]; rollback: Cardano.TxAlonzo[] }>;
```

Comment:
This is ok as it's internal impl details, but maybe separate observables here would be cleaner. What do you think @mirceahasegan ?

```suggestion
  transactionsSource$?: Observable<Cardano.TxAlonzo[]>;
  rollback$?: Observable<Cardano.TxAlonzo>;
```

Reply:
An less invasive to this would be to keep the transactionsSource unchanged (as it is now in master), and use pairwise operator to determine which transactions were removed since the last emission.
The only problem would be if the array of transaction arrays gets really big.
What do you think?

Reply:
Good idea, but I'm also fine with returning it from same function that returns `transactionsSource$`.

> The only problem would be if the array of transaction arrays gets really big.

You can still make it efficient because transactions are supposed to be ordered, so you can just start iterating from end. Not concerned about memory consumption :sweat_smile:

Reply:
> 

If we return it from the same function that returns transactionsSource$, it is easier to emit every time a rollback is detected. The comparing approach is useful only if used externally.

Reply:
Updated in [0286cb7](https://github.com/input-output-hk/cardano-js-sdk/pull/351/commits/0286cb7b5c3844765304acc03f3163345df65fa2)